staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret.go

```diff
@@ -370,25 +370,31 @@ func handleSecretFromFileSources(secret *corev1.Secret, fileSources []string) er
 	return nil
 }
 
-// handleSecretFromEnvFileSource adds the specified env file source information
+// handleSecretFromEnvFileSources adds the specified env files source information
 // into the provided secret
-func handleSecretFromEnvFileSource(secret *corev1.Secret, envFileSource string) error {
-	fileInfo, err := os.Stat(envFileSource)
-	if err != nil {
-		switch err := err.(type) {
-		case *os.PathError:
-			return fmt.Errorf("error reading %s: %v", envFileSource, err.Err)
-		default:
-			return fmt.Errorf("error reading %s: %v", envFileSource, err)
+func handleSecretFromEnvFileSources(secret *corev1.Secret, envFileSources []string) error {
+	for _, envFileSource := range envFileSources {
+		info, err := os.Stat(envFileSource)
+		if err != nil {
+			switch err := err.(type) {
+			case *os.PathError:
+				return fmt.Errorf("error reading %s: %v", envFileSource, err.Err)
+			default:
+				return fmt.Errorf("error reading %s: %v", envFileSource, err)
+			}
+		}
+		if info.IsDir() {
+			return fmt.Errorf("env config file cannot be a directory")
```

Comment:
This error message changed slightly. Was it intentional or ported over from the configmap code?

```suggestion
			return fmt.Errorf("env secret file cannot be a directory")
```

Reply:
Thanks for the quick review! I ported the code from configmap 😅  , will make changes now